src/Libraries/Nop.Services/ExportImport/ImportManager.cs

```diff
@@ -2183,6 +2387,340 @@ public virtual async Task ImportCategoriesFromXlsxAsync(Stream stream)
             throw new ArgumentException(string.Format(await _localizationService.GetResourceAsync("Admin.Catalog.Categories.Import.CategoriesArentImported"), string.Join(", ", categoriesName)));
         }
 
+        /// <summary>
+        /// Import orders from XLSX file
+        /// </summary>
+        /// <param name="stream">Stream</param>
+        /// <returns>A task that represents the asynchronous operation</returns>
+        public virtual async Task ImportOrdersFromXlsxAsync(Stream stream)
+        {
+            using var workbook = new XLWorkbook(stream);
+            // get the first worksheet in the workbook
+            var worksheet = workbook.Worksheets.FirstOrDefault();
+            if (worksheet == null)
+                throw new NopException("No worksheet found");
+
+            var downloadedFiles = new List<string>();
+
+            var metadata = await PrepareImportOrderDataAsync(worksheet);
+
+            //performance optimization, load all orders by guid in one SQL request
+            var allOrdersByGuid = await _orderService.GetOrdersByGuidAsync(metadata.AllOrderGuid.ToArray());
+
+            Order lastLoadedOrder = null;
+
+            for (var iRow = 2; iRow < metadata.EndRow; iRow++)
+            {
+                //imports product attributes
+                if (worksheet.Row(iRow).OutlineLevel != 0)
+                {
+                    if (lastLoadedOrder == null)
+                        continue;
+
+                    metadata.OrderItemManager.ReadFromXlsx(worksheet, iRow, 2);
+
+                    //skip caption row
+                    if (!metadata.OrderItemManager.IsCaption)
+                    {
+                        await ImportOrderItemAsync(metadata.OrderItemManager, lastLoadedOrder);
+                    }
+                    continue;
+                }
+
+                metadata.Manager.ReadFromXlsx(worksheet, iRow);
+
+                var order = metadata.OrderGuidCellNum > 0 ? allOrdersByGuid.FirstOrDefault(p => p.OrderGuid.ToString() == metadata.Manager.GetProperty("OrderGuid").StringValue) : null;

```

Comment:
Let's add the GuidValue property to the PropertyByName class to compare with p.OrderGuid without casting to string